client/src/components/Leaflet.js

```diff
@@ -130,13 +131,14 @@ const Leaflet = ({
         }
         if (m.onMove) {
           marker.on("move", event => m.onMove(event, map))
+          map.on("click", event => marker.setLatLng(event.latlng).update())
```

Comment:
If you do so, if we use this component elsewhere with more that 1 movable marker, all of the markers will be moved on a mouse click. I suggest to define the behaviour uniquely for the Location editor by passing something like a `onClick` prop